tests/test_conformer.py

```diff
@@ -79,3 +87,112 @@ def get_output(shape, norm):
+            block_cfg=ConformerBlockV1Config(
+                ff_cfg=ConformerPositionwiseFeedForwardV1Config(
+                    input_dim=256,
+                    hidden_dim=256,
+                    dropout=0.2,
+                    activation=nn.functional.silu,
+                ),
+                mhsa_cfg=ConformerMHSAV1Config(
+                    input_dim=256,
+                    num_att_heads=4,
+                    att_weights_dropout=0.2,
+                    dropout=0.2,
+                ),
+                conv_cfg=ConformerConvolutionV1Config(
+                    channels=256, kernel_size=9, dropout=0.2, activation=nn.functional.silu, norm=LayerNormNC(256)
+                ),
+            ),
+        )
+
+        class DummyConformerModel(nn.Module):
+            """ """
+
+            def __init__(self, cfg: conformer_config):
+                super().__init__()
+                self.model = ConformerEncoderV1(cfg=cfg)
+
+            def forward(self, input: torch.Tensor, seq_len: torch.Tensor):
+                seq_mask = tensor_mask_from_length(input, seq_len)
+                logits, seq_mask = self.model(input, seq_mask)
+                return logits, seq_mask
+
+        model = DummyConformerModel(cfg=conformer_config)
+        dummy_data = torch.randn(3, 30, 50)
+        dummy_data_len = torch.IntTensor([30, 20, 15])
+        traced_model = torch.jit.trace(model.eval(), example_inputs=(dummy_data, dummy_data_len))
+
+        dummy_data_len_2 = torch.IntTensor([30, 15, 10])
+
+        outputs_normal, _ = model(dummy_data, dummy_data_len)
+        outputs_traced, _ = traced_model(dummy_data, dummy_data_len)
+        # check tracing results in the same outputs
+        assert torch.allclose(outputs_normal, outputs_traced, atol=1e-5)
+
+        export_onnx(
+            traced_model,
+            (dummy_data, dummy_data_len),
+            f=f,
+            verbose=True,
+            input_names=["data", "data_len"],
+            output_names=["classes"],
+            dynamic_axes={
+                "data": {0: "batch", 1: "time"},
+                "data_len": {0: "batch"},
+                "classes": {0: "batch", 1: "time"},
+            },
+        )
+
+        session = ort.InferenceSession(f.name)
+        outputs_onnx = torch.FloatTensor(
+            session.run(None, {"data": dummy_data.numpy(), "data_len": dummy_data_len.numpy()})[0]
+        )
+        outputs_onnx_other = torch.FloatTensor(
+            session.run(None, {"data": dummy_data.numpy(), "data_len": dummy_data_len_2.numpy()})[0]
+        )
+
+        # The default 1e-8 was slightly too strong
+        assert torch.allclose(outputs_normal, outputs_onnx, atol=1e-5)
+        # check that for different lengths we really get a different result
+        assert not torch.allclose(outputs_normal, outputs_onnx_other, atol=1e-5)
+
+        # in the future check with different batching and max size (20)
```

Comment:
is this TODO in this PR or in general?

Reply:
In this PR we can not fix this, so this is a general warning that our model is not batch-size-safe, and to give an idea how to expand the tests once it is.